newIDE/app/src/ObjectsList/index.js

```diff
@@ -222,6 +226,29 @@ export default class ObjectsListContainer extends React.Component<
         }
       }
     );
+
+    if (this.state.makeNewInstanceOnAdd) {
```

Comment:
The ObjectsListContainer is as the name implies something related to the objects list. While it's acceptable to add a prop that is a callback called when an object is added (or when the new object dialog is closed), I think we should avoid at all cost having "Instance" in the objects list.

This is an issue of responsibility: an object list should deal with objects. If we introduce the concept of instance, this make the component harder to reason about - and keeping simple component scoped to a specific thing allow us to reason easily, iterate fastly and build fast :)

Reply:
refactoring this now with a much cleaner way

Reply:
Nice, look forward to see the result :)